datastore/prometheus.go

```diff
@@ -0,0 +1,318 @@
+package datastore
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+	"sync"
+
+	"github.com/ddosify/alaz/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const (
+	projectLabel   = "acorn.io/project-name"
+	appLabel       = "acorn.io/app-public-name"
+	containerLabel = "acorn.io/container-name"
+	accountIdLabel = "acorn.io/account-id"
+)
+
+type PrometheusExporter struct {
+	ctx context.Context
+	reg *prometheus.Registry
+
+	latencyHistogram *prometheus.HistogramVec
+	statusCounter    *prometheus.CounterVec
+
+	podCache *eventCache
+	svcCache *eventCache
+
+	reqChanBuffer chan Request
+	podEventChan  chan Event // *PodEvent
+	svcEventChan  chan Event // *SvcEvent
```

Comment:
The prometheus exporter receives information about HTTP requests, pods, and services through these channels. The aggregator is the component that sends on these channels.

Reply:
I removed the podEventChan and svcEventChan because they were totally unnecessary.